common/flatpak-dir.c

```diff
@@ -10447,17 +10612,24 @@ flatpak_dir_find_remote_related (FlatpakDir *self,
 
               extension_ref = g_build_filename ("runtime", extension, parts[2], branch, NULL);
 
-              if (flatpak_summary_lookup_ref (summary, extension_collection_id, extension_ref, &checksum, NULL))
+              if ((summary == NULL && flatpak_cache_lookup_ref (xa_cache, extension_ref)) ||
+                  (summary != NULL && flatpak_summary_lookup_ref (summary, extension_collection_id, extension_ref, &checksum, NULL)))
```

Comment:
Would it be possible to merge `flatpak_cache_lookup_ref()` and `flatpak_summary_lookup_ref()`? Otherwise you’re going to have to conditionally call both of them from every call site. This should be possible if you can pass the same data structure to both functions (regardless of whether it came from a `summary` file or `ostree-metadata`).

Reply:
Introspection would be terrible. Relying on the data structure being similar enough for `flatpak_variant_bsearch_str()` is pretty yucky too, but will do for a first go at the branch.